.github/workflows/job-checks.yml

```diff
@@ -16,10 +16,19 @@ jobs:
         with:
           python-version: ${{ matrix.python-version }}
           enable-cache: true
+
+      - name: pre-commit cache key
+        run: echo "PY=$(python -VV | sha256sum | cut -d' ' -f1)" >> "$GITHUB_ENV"
+      - uses: actions/cache@v4
+        with:
+          path: ~/.cache/pre-commit
+          key: pre-commit|${{ env.PY }}|${{ hashFiles('.pre-commit-config.yaml') }}
+
       - name: Install dependencies
-        run: uv sync --all-extras --no-extra cu124
+        run: uv sync --frozen --all-extras --no-extra cu124
+
       - name: Run styling check
-        run: uv run --no-sync pre-commit run --all-files
```

Comment:
`--no-sync` was added because it looked like `uv run` is not respecting the extras set before.

basically it ended up installing all the nvidia drivers from pypi.